website/api/controllers/android-proxy/create-android-signup-url.js

```diff
@@ -34,7 +34,50 @@ module.exports = {
     // Check the database for an existing record for this Fleet server.
     let connectionforThisInstanceExists = await AndroidEnterprise.findOne({fleetServerUrl: fleetServerUrl});
     if(connectionforThisInstanceExists){
-      throw 'enterpriseAlreadyExists';
+      // Before throwing conflict, verify the enterprise still exists in Google
+      // If it doesn't exist, clean up the stale proxy record and continue with signup
+      try {
+        let { google } = require('googleapis');
+        let androidmanagement = google.androidmanagement('v1');
+        let googleAuth = new google.auth.GoogleAuth({
+          scopes: ['https://www.googleapis.com/auth/androidmanagement'],
+          credentials: {
+            client_email: sails.config.custom.androidEnterpriseServiceAccountEmailAddress,// eslint-disable-line camelcase
+            private_key: sails.config.custom.androidEnterpriseServiceAccountPrivateKey,// eslint-disable-line camelcase
+          },
+        });
+        let authClient = await googleAuth.getClient();
+        google.options({auth: authClient});
+
+        // Use Google's LIST call to check if enterprise exists
+        let listResponse = await androidmanagement.enterprises.list({
+          projectId: sails.config.custom.androidEnterpriseProjectId,
+        });
+
+        let enterprisesList = listResponse.data;
+        let enterprises = (enterprisesList && enterprisesList.enterprises) ? enterprisesList.enterprises : [];
```

Comment:
This will need to handle multiple pages of results
```suggestion
      let enterprises = [];
      let tokenForNextPageOfEnterprises;
      await sails.helpers.flow.until(async ()=>{
        let listEnterprisesResponse = await androidmanagement.enterprises.list({
          projectId: sails.config.custom.androidEnterpriseProjectId,
          pageSize: 100,
          pageToken: tokenForNextPageOfEnterprises,
        });
        tokenForNextPageOfEnterprises = listEnterprisesResponse.data.nextPageToken;
        enterprises = enterprises.concat(listEnterprisesResponse.data.enterprises);

        if(!listEnterprisesResponse.data.nextPageToken){
          return true;
        }
      });
```